tests/format_test.cc

```diff
@@ -0,0 +1,8 @@
+#include <cris/core/format.h>
+
+#include "gtest/gtest.h"
+
+TEST(FormatTest, FormatTest) {
+    auto output_string = std::format("{} {}", "Hello", 123);
```

Comment:
Readability-wise it's subjective, I know some people hate stream so much, and some even hate printf. To me all have decent readability, sometimes stream is even better because it's linear.

The 20-30x you mentioned is on fp, not general. The table above that 20-30x shows minor diff between fmt and printf.